packages/features/pbac/README.md

```diff
@@ -0,0 +1,228 @@
+# Permission-Based Access Control (PBAC) System
+
+## Overview
+The PBAC system provides fine-grained access control for Cal.com, allowing teams to create custom roles with specific permissions while maintaining backward compatibility with the existing role system.
+
+## Key Concepts
+
+### Permission Format
+Permissions follow the `resource.action` format:
+```typescript
+type Permission = `${Resource}.${Action}`;
+// Example: "eventType.create", "booking.read"
+```
+
+### Role Types
+1. **Default Roles** (MembershipRole) (TODO) - We need to make a list of what permissions these get by default.
+   - OWNER: Full access (`*.*`)
+   - ADMIN: Extensive management permissions
+   - MEMBER: Basic read permissions
+
+2. **Custom Roles**
+   - Team-specific roles with granular permissions
+   - Can be assigned alongside default roles (For now - until we have a seeder or something)
+
+## Usage Guide
+
+### 1. Creating Custom Roles
+```typescript
+const roleService = new RoleService(prisma);
+
+// Create a role with specific permissions
+await roleService.createRole({
+  name: "Event Manager",
+  teamId: 1,
+  permissions: [
+    "eventType.create",
+    "eventType.update",
+    "booking.read",
+    "booking.update"
+  ]
+});
+```
+
+### 2. Assigning Roles to Members
+```typescript
+// Assign a custom role to a team member
+await roleService.assignRoleToMember(roleId, membershipId);
+
+// Remove a custom role
+await roleService.removeRoleFromMember(membershipId);
+```
+
+### 3. Checking Permissions
+```typescript
+const permissionCheck = new PermissionCheckService(roleService);
+
+// Check if a member has a specific permission
+const canCreateEvent = await permissionCheck.hasPermission(
+  {
+    role: MembershipRole.MEMBER,
+    customRoleId: "custom-role-id"
+  },
+  "eventType.create"
+);
```

Comment:
Will likely add a fn overload or a new function here to pass in a userId,teamId and permission